### PR TITLE
Deprecation check for tribe node

### PR DESCRIPTION
**Pull request by AthenaEryma**
Adds a deprecation check for tribe node.

Relates to https://github.com/elastic/elasticsearch/issues/36024 and https://github.com/elastic/elasticsearch/pull/28443

### CODE REVIEW

File: x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java [hunk 1]
```diff
@@ -11,13 +11,29 @@
...
+    static DeprecationIssue tribeNodeCheck(List<NodeInfo> nodeInfos, List<NodeStats> nodeStats) {
+        List<String> nodesFound = nodeInfos.stream()
+            .filter(nodeInfo -> nodeInfo.getSettings().getByPrefix("tribe.").isEmpty() == false)
+            .map(nodeInfo -> nodeInfo.getNode().getName())
+            .collect(Collectors.toList());
+        if (nodesFound.size() > 0) {
+            return new DeprecationIssue(DeprecationIssue.Level.CRITICAL,
+                "Tribe Node removed in favor of Cross Cluster Search",
+                "https://www.elastic.co/guide/en/elasticsearch/reference/master/breaking_70_cluster_changes.html" +
+                    "#_tribe_node_removed",
```
Comment (gwbrown) [inline x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java]: This link should work once https://github.com/elastic/elasticsearch/pull/36239 is merged.

**Comment by gwbrown** [comment]
These test failures appear to be due to the BWC problems around 5.6.15.

@elasticmachine test this please

**Review by s1monw — APPROVED** [review]
LGTM

**Comment by gwbrown** [comment]
Failure does not reproduce locally.
@elasticmachine run gradle build tests 1

**Pull request merged**